### PR TITLE
Add build GitHub Action

### PR DESCRIPTION
**Pull request by pbzweihander**

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: Build container image
+      run: docker build . -t ghcr.io/devsisters/checkpoint:${{ github.sha }}
+    - name: Login to GitHub Container Registry
+      if: github.ref == 'refs/heads/main'
```
Comment (jungnoh) [inline .github/workflows/build.yaml:17]: Is this necessary? We already have `branches: main` on the push trigger.

Comment (pbzweihander) [reply .github/workflows/build.yaml:17]: We also have `pull_request` trigger. This is necessary.

Comment (jungnoh) [reply .github/workflows/build.yaml:17]: Oh I see. How about we add [branch filters](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-including-branches) to `pull_request` instead? This seems like a more consistent pattern for limiting target branches.

Comment (pbzweihander) [reply .github/workflows/build.yaml:17]: I intended to check that all pull requests can be correctly built and selectively (only in the `main` branch) pushed to GHCR.

Comment (jungnoh) [reply .github/workflows/build.yaml:17]: I understand. In this case I would have split the workflow for `push` and `pull_request` entirely, but this seems okay.

**Review by jungnoh — APPROVED** [review]

**Pull request merged**